Beams/Logic/vtkSlicerMlcPositionLogic.h

```diff
@@ -0,0 +1,138 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Portions (c) Copyright Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+==============================================================================*/
+
+// .NAME vtkSlicerMlcPositionLogic - slicer logic class for MLC Position Calculation
+// .SECTION Description
+// This class manages the logic associated with reading, saving,
+// and changing propertied of the volumes
+
+
+#ifndef __vtkSlicerMlcPositionLogic_h
+#define __vtkSlicerMlcPositionLogic_h
+
+#include "vtkSlicerBeamsModuleLogicExport.h"
+
+// Slicer includes
+#include "vtkMRMLAbstractLogic.h"
+
+class vtkPolyData;
+class vtkMRMLMarkupsCurveNode;
+class vtkMRMLRTBeamNode;
+class vtkMRMLTableNode;
+class vtkTable;
+class vtkAlgorithmOutput;
+
+/// \ingroup SlicerRt_QtModules_Beams
+class VTK_SLICER_BEAMS_LOGIC_EXPORT vtkSlicerMlcPositionLogic : public vtkMRMLAbstractLogic {
+public:
+
+  static vtkSlicerMlcPositionLogic *New();
+  vtkTypeMacro( vtkSlicerMlcPositionLogic, vtkMRMLAbstractLogic);
+  void PrintSelf( ostream& os, vtkIndent indent);
+
+  vtkMRMLTableNode* CreateMultiLeafCollimatorTableNodeBoundaryData( bool type, int nofLeafPairs,
+    double leafPairSize, double isocenterOffset);
+
+  /// Update MLC table node boundary data
+  /// Warning: it will also change the table node's name
+  bool UpdateMultiLeafCollimatorTableNodeBoundaryData( vtkMRMLTableNode* mlcTable, 
+    bool type, int nofLeafPairs, double leafPairSize, double isocenterOffset);
+
+  /// Calculate convex hull curve on isocenter plane for MLC position computation
+  vtkMRMLMarkupsCurveNode* CalculatePositionConvexHullCurve( vtkMRMLRTBeamNode* beamNode, 
+    vtkPolyData* targetPoly, bool parallelBeam = true);
+
+  /// Calculate MLC table position for convex hull curve (first pass)
+  bool CalculateMultiLeafCollimatorPosition( vtkMRMLTableNode* mlcTableNode, 
+    vtkMRMLMarkupsCurveNode* curveNode);
+  /// Calculate MLC table position for target polydata (second pass)
+  bool CalculateMultiLeafCollimatorPosition( vtkMRMLRTBeamNode* beamNode, 
+    vtkMRMLTableNode* mlcTableNode, vtkPolyData* targetPoly);
+
+  double CalculateMultiLeafCollimatorPositionArea(vtkMRMLRTBeamNode* beamNode);
```

Comment:
Way more detailed comments please! We need clear and complete documentation, and it would make it much easier for other people but also the future you if you explained what everything does (and why if not obvious)!

Reply:
Descriptions have been added.